### ISSUE TITLE
no memory release and program residues remain active

### ISSUE THREAD

**Issue by ljicsinszky** [JabRef/jabref#4632]
JabRef version <4.3.1> on < Windows 10>

- [x ] I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

<Closing the jabref not only the memory is not released but in the task manager the jabref seems to be running. Even after half an hour. 4 "instances" occupy more than 1 GB. Only manual kill the otherwise not running task releases the memory. If you are doing the same with the development version running directly the jar file, javaw remains in the memory and reserves/uses that. This latter is not always reproducible. Using multiple libraries only the occupied memory increaesed not the repeated appearance of jabref.
 Using the direct running of jar file (431) only Javaplatform binary.../javaw is seen (taskmanager/sysinternals process explorer)>

1. ...open jabref with or without a bib file
2. ...close it
3. ...repeat it a couple of times, in different time and the memory occupancy is accumulated.

<details>
  <summary>Log File</summary>
  
  ```
  Paste an excerpt of your log file here
  ```
</details>
![sysintprocexpl_after_jar_runs](https://user-images.githubusercontent.com/47385109/52339688-17e05f80-2a0e-11e9-8ec6-edd728411c0c.png)
![taskman_after_jar_runs](https://user-images.githubusercontent.com/47385109/52339691-17e05f80-2a0e-11e9-9337-003b5b0b435d.png)
![sysintprocexpl_after_jrexeruns](https://user-images.githubusercontent.com/47385109/52339693-17e05f80-2a0e-11e9-8603-cacb9b6f36d4.png)
![taskman_after_jrexeruns](https://user-images.githubusercontent.com/47385109/52339694-1878f600-2a0e-11e9-9e4f-3cac3c2e2164.png)
![memory_after_jrexeruns](https://user-images.githubusercontent.com/47385109/52339695-1878f600-2a0e-11e9-9dfb-2e18551aca3e.png)
![memory_beforejrruns](https://user-images.githubusercontent.com/47385109/52339696-19118c80-2a0e-11e9-96c1-167cc3ece30a.png)

**Comment by Siedlerchr** [comment]
Thanks for reporting this. Please try the latest development version from https://builds.jabref.org/master

**Comment by ljicsinszky** [comment]
Dear Cristoph,

Thank you for your suggestion.
The bug still exist in JabRef
5.0-dev--snapshot--2019-02-27--master--3cf609cd7 version, see pic below.
Before starting JRdev no java appl was running.

Meantime I have also noticed that if I open a new library within the
running JR(4.3.1) the used memory is accumulated, and after opening/closing
some other bib files the memory usage was over 3GB.
Doing any modifications in the bibfile within the JR the used memory is
increasing. Opening a 8 MB bib file (~3300 records) the memory usage of JR
went up to ~ 700 (!!!) MB, which increased a simple search with 200 MB
(found 560 records). remove the search condition the memory usage was ~80
MB higher than before the searching.
Removing the selected records and saving the residues increased again the
memory usage by ~50 MB.

A simple JR start without opening a bibfile uses ~210 MB (JR4.3.1), while
the current dev version (JabRef
5.0-dev--snapshot--2019-02-27--master--3cf609cd7) uses ~160 MB, and opening
the same ~8 MB file increases the memory use to ~580 MB in the dev version.

I do hope I could help.


Regards,
Laszlo

[image: image.png]

**Comment by tobiasdiez** [comment]
This should be fixed in the latest development version. Could you please check the build from http://builds.jabref.org/master/. Thanks! Please remember to make a backup of your library before trying-out this version.

**Comment by ljicsinszky** [comment]
Dear Tobias,

Thank you for your update and also thank for the work on the JabRef project.
Now, I am using win10 v1903 64 bit and the JabRef versions are also 64-bit
versions with the standard Oracle's Java, 8GB RAM, i5
processor, JabRef-5.0-dev--snapshot--2019-09-16--master--4f7ac7591.jar, 64
bit installer.

 I tried it and I have found the followings:

- Yes, now the used memory by JabRef is completely released by quitting the
program.
- I could not start with my shortcut (the jar file, win10), using a command
prompt I got the following message (it worked with the previous dev
versions)
[image: image.png]
- After that failure to start, I installed the dev version and using the
bat file JabRef has started.

- I have found that the 'accumulation of memory' (opening a new window the
memory use according to the taskmanager is increasing, despite closing the
unused database) is still the property of JabRef but considerably less
expressed. In the previous versions, it started with ~300 MB and every new
database opening practically doubled it). Now the increase is really
moderate.

- I have a very large database, with more than 100000 records, ~130 MB in
size. The previous dev versions could also open it (like the 4.3.1). Now I
got error messages. Even when I tried to open a 3000 records-containing
(~23 MB) truncated database, and the database could not be opened, but
JabRef is still responsive and works.
first error message:
[image: image.png]
second
[image: image.png]

I hope the above comments help you in the development.

Thank you again for your efforts,
Laszlo

**Comment by Siedlerchr** [comment]
Thanks for the feedback. 
Using the jar is no longer possible. We are working on another archive distribution. In the meantime use the installer. 

Could you please attach the images here in the github issue, because when you answer via mail they are not included.

**Comment by tobiasdiez** [comment]
As the original issue is resolved now, I'll close this. Please open another issue for the remaining problem. Thanks.

**Comment by ljicsinszky** [comment]
Dear Tobias,

Thank you for the information.
I am attaching the jar opening problem, although, finally it is not a
problem.
the next two screenshots are the problem of large file opening. First I got
the first picture, and when I closed the window I got the next one.

Good luck,
Laszlo
